### ISSUE TITLE
v4/v5 manually marking factorial experiment failed ("Condition not found")

### ISSUE THREAD

**Issue by danoswaltCL** [CarnegieLearningWeb/UpGrade#1001]
**Version where bug was found:**
"5.0.2"

**Describe the bug**
Manually sending requests to v4 or v5 endpoints for init, assign, and mark. On mark, I get this:

500 Server error when marking factorial experiment:
```
{
  "type": "Condition not found"
}
```

Note, simple experiment marked ok.

**To Reproduce**
In CL staging:

1. init a user
2. assign to "assign-prog" and you should get assignments for a factorial and a simple experiment:

```
[
  {
    "target": "simplify_exponents_2_power",
    "site": "SelectSection",
    "experimentType": "Simple",
    "assignedCondition": {
      "id": "58a69798-c077-49a7-a194-2028a4da0a7c",
      "conditionCode": "Control",
      "payload": {
        "type": "string",
        "value": "control"
      },
      "experimentId": "88ceb386-a54f-4708-b1b5-cea0606ea059"
    }
  },
  {
    "target": "volume_surface_area_cone_vol",
    "site": "SelectSection",
    "experimentType": "Factorial",
    "assignedCondition": {
      "id": "997503c1-3fe6-48c0-aeaa-b197d662305a",
      "conditionCode": "Question Type=Concrete; Motivation=Mindset",
      "payload": null,
      "experimentId": "246f328f-7ebc-46ec-98e6-75cbf68dd04e"
    },
    "assignedFactor": {
      "Question Type": {
        "level": "Concrete",
        "payload": {
          "type": "string",
          "value": "variant"
        }
      },
      "Motivation": {
        "level": "Mindset",
        "payload": {
          "type": "string",
          "value": "variant"
        }
      }
    }
  },
  {
    "target": "volume_surface_area_sphere_vol",
    "site": "SelectSection",
    "experimentType": "Factorial",
    "assignedCondition": {
      "id": "997503c1-3fe6-48c0-aeaa-b197d662305a",
      "conditionCode": "Question Type=Concrete; Motivation=Mindset",
      "payload": null,
      "experimentId": "246f328f-7ebc-46ec-98e6-75cbf68dd04e"
    },
    "assignedFactor": {
      "Question Type": {
        "level": "Concrete",
        "payload": {
          "type": "string",
          "value": "variant"
        }
      },
      "Motivation": {
        "level": "Mindset",
        "payload": {
          "type": "string",
          "value": "variant"
        }
      }
    }
  }
]
```

When I mark with this request, I get the mentioned error:
```
{
  "userId": "{{newUserId}}",
  "status": "condition applied",
  "data": {
    "target": "volume_surface_area_sphere_vol",
    "site": "SelectSection",
    "experimentType": "Factorial",
    "assignedCondition": {
      "id": "997503c1-3fe6-48c0-aeaa-b197d662305a",
      "conditionCode": "Question Type=Concrete; Motivation=Mindset",
      "payload": null,
      "experimentId": "246f328f-7ebc-46ec-98e6-75cbf68dd04e"
    },
    "assignedFactor": {
      "Question Type": {
        "level": "Concrete",
        "payload": {
          "type": "string",
          "value": "variant"
        }
      },
      "Motivation": {
        "level": "Mindset",
        "payload": {
          "type": "string",
          "value": "variant"
        }
      }
    }
  }
}
```

I do not get an error when pasting in the simple experiment assignment object.

**Comment by danoswaltCL** [comment]
Factorial and simple experiment jsons:

```
{"createdAt":"2023-01-31T18:56:24.339Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":18,"id":"246f328f-7ebc-46ec-98e6-75cbf68dd04e","name":"Factorial AA test F1","description":"2x2 Factorial Design, Question Type / Motivation AA Test (no changes to variants)","context":["assign-prog"],"state":"enrolling","startOn":null,"consistencyRule":"individual","assignmentUnit":"individual","postExperimentRule":"continue","enrollmentCompleteCondition":null,"endOn":null,"revertTo":null,"tags":["aa","f1"],"group":null,"conditionOrder":null,"logging":false,"filterMode":"includeAll","backendVersion":"5.0.2","type":"Factorial","partitions":[{"createdAt":"2023-02-08T16:28:47.199Z","updatedAt":"2023-02-08T16:28:47.199Z","versionNumber":1,"id":"6bcadf20-2f2c-47a3-9f07-ef576994465f","twoCharacterId":"8O","site":"SelectSection","target":"volume_surface_area_sphere_vol","description":"","order":1,"excludeIfReached":false,"conditionPayloads":[]},{"createdAt":"2023-02-08T16:28:47.199Z","updatedAt":"2023-02-08T16:28:47.199Z","versionNumber":1,"id":"13eedb16-7769-42ca-9b5d-88e50fa2876c","twoCharacterId":"5O","site":"SelectSection","target":"volume_surface_area_cone_vol","description":"","order":2,"excludeIfReached":false,"conditionPayloads":[]}],"conditions":[{"createdAt":"2023-02-08T16:28:35.823Z","updatedAt":"2023-02-08T16:28:35.823Z","versionNumber":1,"id":"997503c1-3fe6-48c0-aeaa-b197d662305a","twoCharacterId":"Q9","name":"condition 4","description":null,"conditionCode":"condition 4","assignmentWeight":25,"order":1,"levelCombinationElements":[{"createdAt":"2023-09-06T20:22:43.774Z","updatedAt":"2023-09-06T20:22:43.774Z","versionNumber":1,"id":"eef16e55-48e1-44ee-8b95-7c0f60bd0c82","level":{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"513fd318-0b0b-47a6-afb7-76a00a91d36f","name":"Mindset","description":null,"payloadValue":"variant","payloadType":"string","order":null}},{"createdAt":"2023-09-06T20:22:43.774Z","updatedAt":"2023-09-06T20:22:43.774Z","versionNumber":1,"id":"00c17e82-c140-4536-a948-94d2fe397c91","level":{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"9da1ebab-78ef-4207-8241-5af680653abd","name":"Concrete","description":null,"payloadValue":"variant","payloadType":"string","order":null}}]},{"createdAt":"2023-02-08T16:28:35.823Z","updatedAt":"2023-02-08T16:28:35.823Z","versionNumber":1,"id":"2b32ee8b-6918-4ce8-8fd8-1b4678dfca56","twoCharacterId":"AW","name":"condition 3","description":null,"conditionCode":"condition 3","assignmentWeight":25,"order":2,"levelCombinationElements":[{"createdAt":"2023-09-06T20:22:43.774Z","updatedAt":"2023-09-06T20:22:43.774Z","versionNumber":1,"id":"196eed19-1f43-40e7-8017-70db50f8fca5","level":{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"75872a76-ff80-4643-ad11-0b764e37dfda","name":"Control","description":null,"payloadValue":"control","payloadType":"string","order":null}},{"createdAt":"2023-09-06T20:22:43.774Z","updatedAt":"2023-09-06T20:22:43.774Z","versionNumber":1,"id":"cb465cb6-57a7-45c3-9676-5e9cd35ef488","level":{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"9da1ebab-78ef-4207-8241-5af680653abd","name":"Concrete","description":null,"payloadValue":"variant","payloadType":"string","order":null}}]},{"createdAt":"2023-02-08T16:28:35.823Z","updatedAt":"2023-02-08T16:28:35.823Z","versionNumber":1,"id":"d687c7ac-c1ad-4dfe-812d-d7a9790e48c1","twoCharacterId":"VO","name":"condition 2","description":null,"conditionCode":"condition 2","assignmentWeight":25,"order":3,"levelCombinationElements":[{"createdAt":"2023-09-06T20:22:43.774Z","updatedAt":"2023-09-06T20:22:43.774Z","versionNumber":1,"id":"d1d4fbc8-b385-4155-90f8-1f75c56219fa","level":{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"513fd318-0b0b-47a6-afb7-76a00a91d36f","name":"Mindset","description":null,"payloadValue":"variant","payloadType":"string","order":null}},{"createdAt":"2023-09-06T20:22:43.774Z","updatedAt":"2023-09-06T20:22:43.774Z","versionNumber":1,"id":"93a80a10-0e89-45e6-9990-3c74ede962ec","level":{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"547057fa-f6ff-4148-a9ed-df832207c697","name":"Control","description":null,"payloadValue":"control","payloadType":"string","order":null}}]},{"createdAt":"2023-02-08T16:28:35.823Z","updatedAt":"2023-02-08T16:28:35.823Z","versionNumber":1,"id":"c9f73594-317b-4882-9043-0e1ea0498498","twoCharacterId":"8B","name":"condition 1","description":null,"conditionCode":"condition 1","assignmentWeight":25,"order":4,"levelCombinationElements":[{"createdAt":"2023-09-06T20:22:43.774Z","updatedAt":"2023-09-06T20:22:43.774Z","versionNumber":1,"id":"e7f22761-307e-4afd-8cce-ad55779b5f67","level":{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"547057fa-f6ff-4148-a9ed-df832207c697","name":"Control","description":null,"payloadValue":"control","payloadType":"string","order":null}},{"createdAt":"2023-09-06T20:22:43.774Z","updatedAt":"2023-09-06T20:22:43.774Z","versionNumber":1,"id":"3391ef03-73d1-4c5b-b3bd-b02beabfcbf9","level":{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"75872a76-ff80-4643-ad11-0b764e37dfda","name":"Control","description":null,"payloadValue":"control","payloadType":"string","order":null}}]}],"stateTimeLogs":[{"createdAt":"2023-09-07T16:38:08.222Z","updatedAt":"2023-09-07T16:38:08.222Z","versionNumber":1,"id":"0df5cc84-f8e8-42bc-979e-5a43d327647b","fromState":"inactive","toState":"enrolling","timeLog":"2023-09-07T16:38:08.211Z"}],"queries":[{"createdAt":"2023-09-06T20:17:33.321Z","updatedAt":"2023-09-06T20:17:33.321Z","versionNumber":1,"id":"2d9da243-1d1e-4278-9a13-37949ab64a39","name":"Mean problems completed","query":{"operationType":"avg"},"repeatedMeasure":"MOST RECENT","metric":{"createdAt":"2023-06-27T15:50:07.374Z","updatedAt":"2023-06-27T15:50:07.374Z","versionNumber":1,"key":"totalProblemsCompleted","type":"continuous","allowedData":null}},{"createdAt":"2023-09-06T20:17:33.321Z","updatedAt":"2023-09-06T20:17:33.321Z","versionNumber":1,"id":"bb837c43-cf6a-40f4-bb2b-ac8d84a02b36","name":"Max time to complete","query":{"operationType":"max"},"repeatedMeasure":"MOST RECENT","metric":{"createdAt":"2023-06-27T15:50:07.374Z","updatedAt":"2023-06-27T15:50:07.374Z","versionNumber":1,"key":"totalTimeSeconds","type":"continuous","allowedData":null}}],"experimentSegmentInclusion":{"createdAt":"2023-09-06T20:17:33.321Z","updatedAt":"2023-09-06T20:17:33.321Z","versionNumber":1,"segment":{"createdAt":"2023-09-06T20:17:33.354Z","updatedAt":"2023-09-06T20:17:33.354Z","versionNumber":1,"id":"608581ba-95ad-4ee3-ac8c-ff72ad06e387","name":"246f328f-7ebc-46ec-98e6-75cbf68dd04e Inclusion Segment","description":"246f328f-7ebc-46ec-98e6-75cbf68dd04e Inclusion Segment","context":"assign-prog","type":"private","individualForSegment":[],"groupForSegment":[{"createdAt":"2023-09-07T17:11:49.825Z","updatedAt":"2023-09-07T17:11:49.825Z","versionNumber":1,"groupId":"All","type":"All"}],"subSegments":[]}},"experimentSegmentExclusion":{"createdAt":"2023-09-06T20:17:33.321Z","updatedAt":"2023-09-06T20:17:33.321Z","versionNumber":1,"segment":{"createdAt":"2023-09-06T20:17:33.378Z","updatedAt":"2023-09-06T20:17:33.378Z","versionNumber":1,"id":"83d9495a-903d-4a47-806d-556fceaf964d","name":"246f328f-7ebc-46ec-98e6-75cbf68dd04e Exclusion Segment","description":"246f328f-7ebc-46ec-98e6-75cbf68dd04e Exclusion Segment","context":"assign-prog","type":"private","individualForSegment":[],"groupForSegment":[],"subSegments":[]}},"factors":[{"id":"ff80b0ce-0503-4180-b0e0-26d403144e43","name":"Motivation","order":2,"description":null,"levels":[{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"75872a76-ff80-4643-ad11-0b764e37dfda","name":"Control","description":null,"order":null,"payload":{"type":"string","value":"control"}},{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"513fd318-0b0b-47a6-afb7-76a00a91d36f","name":"Mindset","description":null,"order":null,"payload":{"type":"string","value":"variant"}}]},{"id":"637d1d86-cf8f-4736-9de5-6f0fd76a4015","name":"Question Type","order":1,"description":null,"levels":[{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"547057fa-f6ff-4148-a9ed-df832207c697","name":"Control","description":null,"order":null,"payload":{"type":"string","value":"control"}},{"createdAt":"2023-09-07T17:11:49.802Z","updatedAt":"2023-09-07T17:11:49.802Z","versionNumber":1,"id":"9da1ebab-78ef-4207-8241-5af680653abd","name":"Concrete","description":null,"order":null,"payload":{"type":"string","value":"variant"}}]}],"conditionPayloads":[]}
```

```
{"createdAt":"2022-10-27T19:11:03.552Z","updatedAt":"2023-09-07T17:15:39.087Z","versionNumber":10,"id":"88ceb386-a54f-4708-b1b5-cea0606ea059","name":"AA2","description":"Conducting AA2 experiment post Factorial redesign","context":["assign-prog"],"state":"enrolling","startOn":null,"consistencyRule":"individual","assignmentUnit":"individual","postExperimentRule":"continue","enrollmentCompleteCondition":null,"endOn":null,"revertTo":null,"tags":["aa"],"group":null,"conditionOrder":null,"logging":false,"filterMode":"includeAll","backendVersion":"5.0.2","type":"Simple","partitions":[{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"77edee94-7a3f-4d56-bd78-f1ad2f936703","twoCharacterId":"C6","site":"SelectSection","target":"simplify_exponents_2_power","description":"","order":1,"excludeIfReached":false}],"conditions":[{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"58a69798-c077-49a7-a194-2028a4da0a7c","twoCharacterId":"IG","name":"","description":null,"conditionCode":"Control","assignmentWeight":50,"order":2,"conditionPayloads":[{"createdAt":"2023-09-07T17:11:01.972Z","updatedAt":"2023-09-07T17:11:01.972Z","versionNumber":1,"id":"1fc7339f-3d91-4874-8445-80c528b315c6","payloadValue":"control","payloadType":"string"},{"createdAt":"2023-09-07T17:09:25.512Z","updatedAt":"2023-09-07T17:09:25.512Z","versionNumber":1,"id":"44b9b900-1557-4302-aa63-3259dd9cfa67","payloadValue":"control","payloadType":"string"}],"levelCombinationElements":[]},{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"7e63ca83-3b36-4ab9-957a-582b11779c24","twoCharacterId":"LD","name":"","description":null,"conditionCode":"Variant","assignmentWeight":50,"order":1,"conditionPayloads":[{"createdAt":"2023-09-07T17:11:01.972Z","updatedAt":"2023-09-07T17:11:01.972Z","versionNumber":1,"id":"a21873e6-ba20-48c1-abd2-3705a10a175e","payloadValue":"variant","payloadType":"string"},{"createdAt":"2023-09-07T17:09:25.512Z","updatedAt":"2023-09-07T17:09:25.512Z","versionNumber":1,"id":"b652debc-f6c5-4ec1-847e-cf664686541d","payloadValue":"variant","payloadType":"string"}],"levelCombinationElements":[]}],"stateTimeLogs":[{"createdAt":"2023-09-07T17:15:39.091Z","updatedAt":"2023-09-07T17:15:39.091Z","versionNumber":1,"id":"fc2ca864-5fd9-4e81-8f9c-571cd80a78e1","fromState":"inactive","toState":"enrolling","timeLog":"2023-09-07T17:15:39.083Z"}],"queries":[{"createdAt":"2023-09-07T17:06:19.495Z","updatedAt":"2023-09-07T17:06:19.495Z","versionNumber":1,"id":"10fe673a-d612-4307-a205-1f0dc5d86c7b","name":"Percent Graduated","query":{"operationType":"percentage"},"repeatedMeasure":"MOST RECENT","metric":{"createdAt":"2023-06-27T15:50:07.374Z","updatedAt":"2023-06-27T15:50:07.374Z","versionNumber":1,"key":"masteryWorkspace@__@absolute_value_plot_equality@__@workspaceCompletionStatus","type":"categorical","allowedData":["GRADUATED","PROMOTED"]}},{"createdAt":"2023-09-07T17:06:19.495Z","updatedAt":"2023-09-07T17:06:19.495Z","versionNumber":1,"id":"e59d80e2-a4bc-4349-8882-2521094a232b","name":"Mean Workspace Time Seconds","query":{"operationType":"avg"},"repeatedMeasure":"MOST RECENT","metric":{"createdAt":"2023-06-27T15:50:07.374Z","updatedAt":"2023-06-27T15:50:07.374Z","versionNumber":1,"key":"masteryWorkspace@__@absolute_value_plot_equality@__@timeSeconds","type":"continuous","allowedData":null}},{"createdAt":"2023-09-07T17:06:19.495Z","updatedAt":"2023-09-07T17:06:19.495Z","versionNumber":1,"id":"b2f9a88a-862c-4cc2-8971-e19ce9126619","name":"Median Total Problems Completed","query":{"operationType":"median"},"repeatedMeasure":"MOST RECENT","metric":{"createdAt":"2023-06-27T15:50:07.374Z","updatedAt":"2023-06-27T15:50:07.374Z","versionNumber":1,"key":"totalProblemsCompleted","type":"continuous","allowedData":null}}],"experimentSegmentInclusion":{"createdAt":"2023-09-07T17:06:19.495Z","updatedAt":"2023-09-07T17:06:19.495Z","versionNumber":1,"segment":{"createdAt":"2023-09-07T17:06:19.575Z","updatedAt":"2023-09-07T17:06:19.575Z","versionNumber":1,"id":"78babbfb-171a-4be8-85c9-c666a7829755","name":"88ceb386-a54f-4708-b1b5-cea0606ea059 Inclusion Segment","description":"88ceb386-a54f-4708-b1b5-cea0606ea059 Inclusion Segment","context":"assign-prog","type":"private","individualForSegment":[],"groupForSegment":[{"createdAt":"2023-09-07T17:12:41.973Z","updatedAt":"2023-09-07T17:12:41.973Z","versionNumber":1,"groupId":"All","type":"All"}],"subSegments":[]}},"experimentSegmentExclusion":{"createdAt":"2023-09-07T17:06:19.495Z","updatedAt":"2023-09-07T17:06:19.495Z","versionNumber":1,"segment":{"createdAt":"2023-09-07T17:06:19.652Z","updatedAt":"2023-09-07T17:06:19.652Z","versionNumber":1,"id":"c279439f-1675-4983-9f75-43c167450ad6","name":"88ceb386-a54f-4708-b1b5-cea0606ea059 Exclusion Segment","description":"88ceb386-a54f-4708-b1b5-cea0606ea059 Exclusion Segment","context":"assign-prog","type":"private","individualForSegment":[],"groupForSegment":[],"subSegments":[]}},"factors":[],"conditionPayloads":[{"createdAt":"2023-09-07T17:09:25.512Z","updatedAt":"2023-09-07T17:09:25.512Z","versionNumber":1,"id":"b652debc-f6c5-4ec1-847e-cf664686541d","parentCondition":{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"7e63ca83-3b36-4ab9-957a-582b11779c24","twoCharacterId":"LD","name":"","description":null,"conditionCode":"Variant","assignmentWeight":50,"order":1},"decisionPoint":{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"77edee94-7a3f-4d56-bd78-f1ad2f936703","twoCharacterId":"C6","site":"SelectSection","target":"simplify_exponents_2_power","description":"","order":1,"excludeIfReached":false},"payload":{"type":"string","value":"variant"}},{"createdAt":"2023-09-07T17:11:01.972Z","updatedAt":"2023-09-07T17:11:01.972Z","versionNumber":1,"id":"a21873e6-ba20-48c1-abd2-3705a10a175e","parentCondition":{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"7e63ca83-3b36-4ab9-957a-582b11779c24","twoCharacterId":"LD","name":"","description":null,"conditionCode":"Variant","assignmentWeight":50,"order":1},"decisionPoint":{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"77edee94-7a3f-4d56-bd78-f1ad2f936703","twoCharacterId":"C6","site":"SelectSection","target":"simplify_exponents_2_power","description":"","order":1,"excludeIfReached":false},"payload":{"type":"string","value":"variant"}},{"createdAt":"2023-09-07T17:09:25.512Z","updatedAt":"2023-09-07T17:09:25.512Z","versionNumber":1,"id":"44b9b900-1557-4302-aa63-3259dd9cfa67","parentCondition":{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"58a69798-c077-49a7-a194-2028a4da0a7c","twoCharacterId":"IG","name":"","description":null,"conditionCode":"Control","assignmentWeight":50,"order":2},"decisionPoint":{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"77edee94-7a3f-4d56-bd78-f1ad2f936703","twoCharacterId":"C6","site":"SelectSection","target":"simplify_exponents_2_power","description":"","order":1,"excludeIfReached":false},"payload":{"type":"string","value":"control"}},{"createdAt":"2023-09-07T17:11:01.972Z","updatedAt":"2023-09-07T17:11:01.972Z","versionNumber":1,"id":"1fc7339f-3d91-4874-8445-80c528b315c6","parentCondition":{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"58a69798-c077-49a7-a194-2028a4da0a7c","twoCharacterId":"IG","name":"","description":null,"conditionCode":"Control","assignmentWeight":50,"order":2},"decisionPoint":{"createdAt":"2022-10-21T13:27:40.847Z","updatedAt":"2022-10-21T13:27:40.847Z","versionNumber":1,"id":"77edee94-7a3f-4d56-bd78-f1ad2f936703","twoCharacterId":"C6","site":"SelectSection","target":"simplify_exponents_2_power","description":"","order":1,"excludeIfReached":false},"payload":{"type":"string","value":"control"}}]}
```

**Comment by danoswaltCL** [comment]
Update: Further looking in db, the conditions are indeed not there, so this may be more related to conditions not getting inserted into the database when imported when the experiment is factorial.

See also #1002, maybe factorial experiment jsons on import or CRUD operations to `/experiments` are not getting processed as expected.

**Comment by ppratikcr7** [comment]
@danoswaltCL Thanks for the details. When I checked this factorial experiment on staging as well as locally via import, I found that somehow the conditions are not being stored in `Factor1=Level1; Factor2=Level2` format. In the backend we have conditionCodes stored as `condition 1`, `condition 2`, `condition 3`, `condition 4`. If we hit mark call with these conditionCodes, we wont see the error and mark would be successful. Also, these conditions are not getting populated in the stepper and details page as shown below. 

Now, the only question is that, how at the first place we stored these condition names. As, we don't allow to edit the condition names from stepper. Is this some old experiment that was imported with these condition names? Or has someone made edits in staging DB data directly?

<img width="1169" alt="Screenshot 2023-09-14 at 1 26 17 PM" src="https://github.com/CarnegieLearningWeb/UpGrade/assets/33730817/77bec6fc-f5f8-4c80-a3c5-c47055c06d1c">
<img width="895" alt="Screenshot 2023-09-14 at 5 25 48 PM" src="https://github.com/CarnegieLearningWeb/UpGrade/assets/33730817/70ca5e19-110f-48ac-b4da-18eaf8b379d3">
<img width="712" alt="Screenshot 2023-09-14 at 5 32 52 PM" src="https://github.com/CarnegieLearningWeb/UpGrade/assets/33730817/5b1fb873-b52e-445d-8b11-6a9f8b440c62">
<img width="1179" alt="Screenshot 2023-09-14 at 5 36 37 PM" src="https://github.com/CarnegieLearningWeb/UpGrade/assets/33730817/9e309168-47fe-406b-9cf8-fd469b20dcfa">
